functions/error-reporting.ts

```diff
@@ -0,0 +1,43 @@
+import process from 'process'
+
+import Bugsnag from '@bugsnag/js'
+import { Handler } from '@netlify/functions'
+
+Bugsnag.start({
+  apiKey: `${process.env.NETLIFY_BUGSNAG_API_KEY}`,
+})
+
+export const handler: Handler = async ({ body }) => {
+  try {
+    if (typeof body !== 'string') {
+      return { statusCode: 200 }
+    }
+    const {
+      cause,
+      cliVersion,
+      message,
+      name,
+      nodejsVersion,
+      osName,
+      severity = 'error',
+      stack,
+      user,
+    } = JSON.parse(body)
+    Bugsnag.notify({ name, message, stack, cause }, (event) => {
+      event.setUser(user.id, user.email, user.name)
```

Comment:
I was unsure if we can do this with all params being undefined, but it seems to be okay according to their types: https://github.com/bugsnag/bugsnag-js/blob/2cfdefb93805673871a676ecbe08155675d046fd/packages/core/types/event.d.ts#L42

Reply:
yep they all accept undefined ;)